qa/integration/specs/monitoring_api_spec.rb

```diff
@@ -286,6 +287,46 @@
     end
   end
 
+  it "should retrieve plugin level flow metrics" do
+    logstash_service = @fixture.get_service("logstash")
+    logstash_service.start_with_stdin(plugins_config)
+    logstash_service.wait_for_logstash
+    number_of_events.times {
+      logstash_service.write_to_stdin("Testing plugin-level flow metrics")
+      sleep(1)
+    }
+
+    Stud.try(max_retry.times, [StandardError, RSpec::Expectations::ExpectationNotMetError]) do
+      # node_stats can fail if the stats subsystem isn't ready
+      result = logstash_service.monitoring_api.node_stats rescue nil
+      expect(result).not_to be_nil
+      # we use fetch here since we want failed fetches to raise an exception
+      # and trigger the retry block
+      expect(result).to include('pipelines' => hash_including('main' => hash_including('plugins' => hash_including('inputs'))))
+      expect(result).to include('pipelines' => hash_including('main' => hash_including('plugins' => hash_including('filters'))))
+      expect(result).to include('pipelines' => hash_including('main' => hash_including('plugins' => hash_including('outputs'))))
```

Comment:
These could be condensed:

```suggestion
      expect(result).to include('pipelines' => hash_including('main' => hash_including('plugins' => hash_including('inputs', 'filters', 'outputs'))))
```